rbi/core/thread.rbi

```diff
@@ -1015,7 +1030,13 @@ class Thread < Object
   #   # all exceptions inherited from Exception are prohibited.
   # }
   # ```
-  sig {params(hash: T.untyped, block: T.proc.returns(T.untyped)).returns(T.untyped)}
+  sig do
+    type_parameters(:U).params(
+      hash: T::Hash[T::Class[T.anything], Symbol],
```

Comment:
Should this be `T.class_of(Exception)` instead of `T::Class[T.anything]`? What would it mean to have a class here that's not an exception?

Reply:
Ruby doesn't validate it and accepts anything here but I can't think of any scenario where it would do anything. Should I switch it to `T.class_of(Exception)`?

Reply:
I would probably prefer `T.class_of(Exception)` to alert someone that they're probably doing something wrong.

Reply:
Just tested this code and it does work:


```
t = Thread.new do
  Thread.handle_interrupt(Object => :never) do
    sleep(5)
  end
end
```

It's perhaps odd but is functional.

Reply:
This also works at runtime:

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Asig%20%7B%20params%28opts%3A%20T%3A%3AHash%5BSymbol%2C%20Integer%5D%29.void%20%7D%0Adef%20example%28opts%29%0A%20%20opts%5B%22not%20a%20symobl%22%5D%0Aend)

```ruby
# typed: true
extend T::Sig

sig { params(opts: T::Hash[Symbol, Integer]).void }
def example(opts)
  opts["not a symobl"]
end
```

but despite the fact that no runtime exception is raised, Sorbet reports a type error because chances are that the user is confused.

I don't care that much about this particular signature and if you'd like we can keep it at `T::Class[T.anything]`, but I think that in general we use types to prevent people from accidentally making mistakes, not necessarily to allow everything that the runtime allows.

Reply:
Cool, sounds good to me. Updated now.